packages/medusa/src/interfaces/abstract-parser.ts

```diff
@@ -0,0 +1,30 @@
+import { ParseConfig } from "papaparse"
+
+export interface IParser<TResult> {
+  parse(path: string): Promise<TResult[]>
+}
+
+export type LineContext = {
+  lineNumber: number
+}
+
+export type ParserOptions = ParseConfig
```

Comment:
question: does that type make sense? We are re exporting an existing type

Reply:
initially, I made the type to decouple the `ParseConfig` from the underlying csv parsing library (Papaparse), but we probably still need to figure out a common interface between the different libraries. wdyt?

Reply:
Yes I understand but we need more to take some step back to be able to know what we need. Let it that way for now and will see with time